pkg/client/logs.go

```diff
@@ -180,8 +181,47 @@ type logStreamer struct {
 	client             kubernetes.Interface
 }
 
+func isContainerRunning(statuses *[]v1.ContainerStatus, containerName string) bool {
+	for _, cs := range *statuses {
+		if cs.Name == containerName && cs.State.Running != nil {
+			return true
+		}
+	}
+	return false
+}
+
+func (l *logStreamer) waitForContainerRunning() error {
+	backoffSeconds := 1 * time.Second
+	for {
+		pod, err := l.client.CoreV1().Pods(l.ns).Get(l.pod, metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("failed to get pod [%s/%s]", l.ns, l.pod)
+		}
+
+		for _, cs := range pod.Status.ContainerStatuses {
+			if cs.Name == l.container && cs.State.Running != nil {
+				return nil
+			}
+			if isContainerRunning(&pod.Status.ContainerStatuses, l.container) {
+				return nil
+			}
+		}
+
+		//TODO: add a time out
```

Comment:
I dont think we need a timeout since this only occurs when they want to follow the logs which doesn't implicily have any sort of ; if they are tailing the logs itself doesnt have a timeout either.